src/webview/webViewEventHandlers.js

```diff
@@ -199,9 +199,18 @@ export const handleMessageListEvent = (props: Props, _: GetText, event: MessageL
       markRead(props, event);
       break;
 
-    case 'avatar':
-      props.dispatch(navigateToAccountDetails(event.fromEmail));
+    case 'avatar': {
+      const userId = parseInt(event.fromUserId, 10);
+      if (Number.isNaN(userId)) {
+        logging.error("WebViewEventHandlers: Failed to parse 'fromUserId' string to integer", {
+          fromUserId: event.fromUserId,
+        });
+        showToast(_('The profile for this user could not be found.'));
+      } else {
```

Comment:
If this error condition ever happens, it's a bug purely internal to our code, and it's not one that even relies on any subtle invariants of our big data structures -- there's a handful of steps from our templates, via the click handler in `js.js`, to here, and this bug would mean one of those somehow fumbled the data.

The usual way we handle that kind of potential bug is by just throwing an exception. I think that's the most appropriate thing to do here, too. As you can see, trying to do this style of more-verbose error handling makes for quite a bit more code to read :wink: 

(This style is exactly the right kind of approach when parsing data we get from the network! But here we're inside the ["soft center"](https://github.com/zulip/zulip-mobile/blob/master/docs/architecture/crunchy-shell.md) of the app, and so we should take advantage of that.)